File: content-service/internal/entity/category.go
package entity

// Category entity
// ...existing code...

type Category struct {
	Base
	Name      string     `gorm:"type:varchar(100);not null;unique" json:"name"`
	Slug      string     `gorm:"type:varchar(150);not null;unique;index" json:"slug"`
	Showcases []Showcase `gorm:"foreignKey:CategoryID" json:"showcases,omitempty"`
}

// CategoryDTO untuk response tanpa deleted_at
// DTO ini digunakan di handler

type CategoryDTO struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	Slug      string `json:"slug"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

File: frontend/src/types/category.ts
/**
 * Category entity (DTO from backend)
 * createdAt & updatedAt opsional/nullable agar robust jika field tidak ada di response
 */
export interface Category {
	id: string;
	name: string;
	slug: string;
	createdAt?: string | null;
	updatedAt?: string | null;
}
